server/src/test/java/org/elasticsearch/snapshots/BlobStoreIncrementalityIT.java

```diff
@@ -0,0 +1,415 @@
+import org.elasticsearch.repositories.RepositoryData;
+import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
+import org.elasticsearch.repositories.blobstore.ChecksumBlobStoreFormat;
+import org.elasticsearch.test.IntegTestCase;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakGroup;
+
+import io.crate.action.FutureActionListener;
+import io.crate.testing.UseRandomizedSchema;
+
+// tests needs to be reworked when: https://github.com/crate/crate/issues/12598 is implemented
+// in order to match the original test from es (restore table with different names
+// https://github.com/elastic/elasticsearch/commit/5b9864db2c318b5fd9699075b6e8e3ccab565675
+@IntegTestCase.ClusterScope(scope = IntegTestCase.Scope.TEST, numDataNodes = 0)
+@ThreadLeakGroup
+public class BlobStoreIncrementalityIT extends AbstractSnapshotIntegTestCase {
+
+    /**
+     * Original defined at BlobStoreRepository, but removed as not used.
+     * Used by {@link #assertTwoIdenticalShardSnapshots(String, String, String, String)}
+     */
+    private static final ChecksumBlobStoreFormat<BlobStoreIndexShardSnapshot> INDEX_SHARD_SNAPSHOT_FORMAT =
+        new ChecksumBlobStoreFormat<>(
+            SNAPSHOT_CODEC, SNAPSHOT_NAME_FORMAT,
+            BlobStoreIndexShardSnapshot::fromXContent,
+            NamedXContentRegistry.EMPTY,
+            false
+        );
+
+    @ClassRule
+    public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal))
+            .put("path.repo", TEMP_FOLDER.getRoot().getAbsolutePath())
+            .build();
+    }
+
+    @UseRandomizedSchema(random = false)
+    @Test
+    public void testIncrementalBehaviorOnPrimaryFailover() throws Exception {
+        internalCluster().startMasterOnlyNode();
+        final String primaryNode = internalCluster().startDataOnlyNode();
+        final String indexName = "test_index";
+        execute("CREATE TABLE " + indexName + "(a string) " +
+                "CLUSTERED INTO 1 SHARDS WITH(number_of_replicas=1,\"unassigned.node_left.delayed_timeout\"=0)");
+        ensureYellow();
+        final String newPrimary = internalCluster().startDataOnlyNode();
+        ensureGreen();
+
+        logger.info("--> adding some documents to test index");
+        int documentCountOriginal = 0;
+        for (int j = 0; j < randomIntBetween(1, 10); ++j) {
+            for (int i = 0; i < scaledRandomIntBetween(1, 100); ++i) {
+                String data = "foo" + j + "_bar" + i;
+                execute("INSERT INTO " + indexName + "(a) VALUES(?)", new Object[]{data});
+                documentCountOriginal++;
+            }
+        }
```

Comment:
Is this important that these are single inserts? Otherwise could use bulk inserts to speed up the test case.

Reply:
You're right, I can change to bulk insert in this test.